### PR TITLE
Context menu API

### PR DESCRIPTION
**Pull request by paulrouget**
This adds an API for Servo internals to request a context menu to the embedder, along an implement for the UWP port.

### CODE REVIEW

**Comment by jdm** [comment]
```
error[E0004]: non-exhaustive patterns: `ShowContextMenu(_, _)` not covered
   --> ports/glutin/browser.rs:280:19
    |
280 |             match msg {
    |                   ^^^ pattern `ShowContextMenu(_, _)` not covered
    |
    = help: ensure that all possible cases are being handled, possibly by adding wildcards or more match arms
error: aborting due to previous error
```

File: ports/libsimpleservo/capi/src/lib.rs [hunk 1]
```diff
@@ -874,4 +902,16 @@ impl HostTrait for HostCallbacks {
...
+    fn show_context_menu(&self, items: Vec<String>) {
+        debug!("show_context_menu");
+        let size = items.len() as u32;
+        let cstrs: Vec<CString> = items
+            .into_iter()
+            .map(|i| CString::new(i).expect("Can't create string"))
+            .collect();
+        let ptrs: Vec<*const c_char> = cstrs.iter().map(|cstr| cstr.as_ptr()).collect();
+        (self.0.show_context_menu)(ptrs.as_ptr(), size);
+        let _ = cstrs; // Don't drop these too early
```
Comment (jdm) [inline ports/libsimpleservo/capi/src/lib.rs]: This line doesn't look like it should actually be necessary - won't cstrs last until the end of function by default?

**Review by jdm — APPROVED** [review]

File: ports/libsimpleservo/capi/src/lib.rs (hunk 1 above)
Comment (paulrouget) [reply ports/libsimpleservo/capi/src/lib.rs]: Hmm… I tried again, and indeed `cstrs` is not dropped early.
For some reason this was necessary at some point, not sure why.

I removed the line.

**Comment by paulrouget** [comment]
@bors-servo retry

- #24355

**Comment by CYBAI** [comment]
@bors-servo retry

- #25513

**Pull request merged**